File: SDKs/amazonaws.com/glue/2017-03-31/python/src/sdk/models/operations/getworkflowrunproperties.py
from dataclasses import dataclass, field
from typing import Any,Enum,Optional
from sdk.models import shared

class GetWorkflowRunPropertiesXAmzTargetEnum(str, Enum):
    AWS_GLUE_GET_WORKFLOW_RUN_PROPERTIES = "AWSGlue.GetWorkflowRunProperties"


@dataclass
class GetWorkflowRunPropertiesHeaders:
    x_amz_algorithm: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-Algorithm' }})
    x_amz_content_sha256: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-Content-Sha256' }})
    x_amz_credential: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-Credential' }})
    x_amz_date: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-Date' }})
    x_amz_security_token: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-Security-Token' }})
    x_amz_signature: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-Signature' }})
    x_amz_signed_headers: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-SignedHeaders' }})
    x_amz_target: GetWorkflowRunPropertiesXAmzTargetEnum = field(default=None, metadata={'header': { 'field_name': 'X-Amz-Target' }})
    

@dataclass
class GetWorkflowRunPropertiesRequest:
    headers: GetWorkflowRunPropertiesHeaders = field(default=None)
    request: shared.GetWorkflowRunPropertiesRequest = field(default=None, metadata={'request': { 'media_type': 'application/json' }})
    

@dataclass
class GetWorkflowRunPropertiesResponse:
    content_type: str = field(default=None)
    entity_not_found_exception: Optional[Any] = field(default=None)
    get_workflow_run_properties_response: Optional[shared.GetWorkflowRunPropertiesResponse] = field(default=None)
    internal_service_exception: Optional[Any] = field(default=None)
    invalid_input_exception: Optional[Any] = field(default=None)
    operation_timeout_exception: Optional[Any] = field(default=None)
    status_code: int = field(default=None)
    

File: SDKs/amazonaws.com/glue/2017-03-31/go/pkg/models/operations/getworkflowrunproperties.go
package operations

import (
	"openapi/pkg/models/shared"
)

type GetWorkflowRunPropertiesXAmzTargetEnum string

const (
	GetWorkflowRunPropertiesXAmzTargetEnumAwsGlueGetWorkflowRunProperties GetWorkflowRunPropertiesXAmzTargetEnum = "AWSGlue.GetWorkflowRunProperties"
)

type GetWorkflowRunPropertiesHeaders struct {
	XAmzAlgorithm     *string                                `header:"style=simple,explode=false,name=X-Amz-Algorithm"`
	XAmzContentSha256 *string                                `header:"style=simple,explode=false,name=X-Amz-Content-Sha256"`
	XAmzCredential    *string                                `header:"style=simple,explode=false,name=X-Amz-Credential"`
	XAmzDate          *string                                `header:"style=simple,explode=false,name=X-Amz-Date"`
	XAmzSecurityToken *string                                `header:"style=simple,explode=false,name=X-Amz-Security-Token"`
	XAmzSignature     *string                                `header:"style=simple,explode=false,name=X-Amz-Signature"`
	XAmzSignedHeaders *string                                `header:"style=simple,explode=false,name=X-Amz-SignedHeaders"`
	XAmzTarget        GetWorkflowRunPropertiesXAmzTargetEnum `header:"style=simple,explode=false,name=X-Amz-Target"`
}

type GetWorkflowRunPropertiesRequest struct {
	Headers GetWorkflowRunPropertiesHeaders
	Request shared.GetWorkflowRunPropertiesRequest `request:"mediaType=application/json"`
}

type GetWorkflowRunPropertiesResponse struct {
	ContentType                      string
	EntityNotFoundException          *interface{}
	GetWorkflowRunPropertiesResponse *shared.GetWorkflowRunPropertiesResponse
	InternalServiceException         *interface{}
	InvalidInputException            *interface{}
	OperationTimeoutException        *interface{}
	StatusCode                       int64
}
